### PR TITLE
fix: guide lines ignored on loading from old files

### PR DESCRIPTION
**Pull request by rodolforg**
Due to changes in b172e3771a131dbd26879aae334b84ac26bcb6e6 (#2777)

### CODE REVIEW

File: synfig-core/src/synfig/loadcanvas.cpp [hunk 1]
```diff
@@ -3359,7 +3360,46 @@ CanvasParser::parse_canvas(xmlpp::Element *element,Canvas::Handle parent,bool in
...
-				canvas->set_meta_data(child->get_attribute("name")->get_value(),content);
+
+				// Guide lines storage changed:
+				// Before:
+				//   guide_x -> list of X positions of vertical guide lines (separated by spaces)
+				//   guide_y -> list of Y positions of horizontal guide lines (separated by spaces)
+				// Now:
+				//   guide -> list of X*Y*angle coordinates of guide lines (data from same guide is separated by star *) (items separated by spaces)
+				if (meta_name == "guide_x") {
+					std::string new_content;
```
Comment (ice0) [inline synfig-core/src/synfig/loadcanvas.cpp:3371]: I suggest to extract this code to the separate function, something like this:
```
void upgrade_guide_metadata(std::string& content, const std::string& metadata, bool is_x_guide) {
	std::string new_content;
	std::istringstream iss(content);
	std::string item;
	while (std::getline(iss, item, ' ')) {
		if (!item.empty()) {
			if (is_x_guide) {
				new_content += item + "*0*1.57079633 ";
			} else {
				new_content += "0*" + item + "*0 ";
			}
		}
	}

	content = new_content + " " + metadata;
	if (content.back() == ' ')
		content.pop_back();

}
```

File: synfig-core/src/synfig/loadcanvas.cpp [hunk 2]
```diff
@@ -3359,7 +3360,46 @@ CanvasParser::parse_canvas(xmlpp::Element *element,Canvas::Handle parent,bool in
...
+					std::istringstream iss(content);
+					std::string item;
+					while (std::getline(iss, item, ' ')) {
+						if (!item.empty()) {
+							new_content += item + "*0*1.57079633 ";
```
Comment (ice0) [inline synfig-core/src/synfig/loadcanvas.cpp:3376]: `1.57079633` - magic number.
Please add explanation in comment.

File: synfig-core/src/synfig/loadcanvas.cpp [hunk 3]
```diff
@@ -3359,7 +3360,46 @@ CanvasParser::parse_canvas(xmlpp::Element *element,Canvas::Handle parent,bool in
...
+					std::string new_content;
+					std::istringstream iss(content);
+					std::string item;
+					while (std::getline(iss, item, ' ')) {
+						if (!item.empty()) {
+							new_content += "0*" + item + "*0 ";
+						}
+					}
+
+					content = new_content + " " +canvas->get_meta_data("guide");
```
Comment (ice0) [inline synfig-core/src/synfig/loadcanvas.cpp:3396]: > content = new_content + " " +canvas->get_meta_data("guide");

Do we really need to add a space here? 
If we made the conversion, then we already have them, if not, then it seems like we don’t need them.

Comment (rodolforg) [reply synfig-core/src/synfig/loadcanvas.cpp]: You're right! Done

File: synfig-core/src/synfig/loadcanvas.cpp (hunk 2 above)
Comment (rodolforg) [reply synfig-core/src/synfig/loadcanvas.cpp]: Done!

File: synfig-core/src/synfig/loadcanvas.cpp (hunk 1 above)
Comment (rodolforg) [reply synfig-core/src/synfig/loadcanvas.cpp]: Done!

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
> I now wonder if this code should be moved to `App::open*()`, as this (any?) metadata is only used by GUI.

I think it's better to keep all the conversion logic in one place. But if there will be a lot of conversions just for the GUI, then yes, it makes sense.

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**